pkg/controller/controller.go

```diff
@@ -0,0 +1,175 @@
+package controller
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	certv1beta1 "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	configNamespace    = "openshift-config-managed"
+	kubeletCAConfigMap = "csr-controller-ca"
+)
+
+// MachineApproverReconciler reconciles a machine-approver  object
+type MachineApproverReconciler struct {
+	client.Client
+	Mgr    ctrl.Manager
```

Comment:
We should probably copy other patterns and only embed the bit that we need, I think just the rest config here